src/akamai/js/akamaiHD_flash.js

```diff
@@ -451,6 +488,95 @@ require("../../../html5-common/js/utils/constants.js");
 
     // Receives a callback from Flash
     onCallback = _.bind(function(data) {
+      console.log("[AKAMIHD]:onCallback: ", data);
+      var eventtitle =" ";
+
+      for(var key in data) {
+        if (key == "eventtype") {
+             eventtitle = data[key];
+        }
+        else if (key =="eventObject") {
+              eventData = data[key];
+        }
+      }
+      if (eventData != null) {
+        for (var item in eventData)
+        {
+          if (item == "currentTime") {
+                currentTime = eventData[item];
+          }
+          else if (item == "buffer") {
+                buffer = eventData[item];
+          }
+          else if (item == "duration") {
+                totalTime =eventData[item];
+          }
+          else if (item == "seekRange_start") {
+                seekRange_start = eventData[item];
+          }
+          else if (item == "seekRange_end") {
+                seekRange_end = eventData[item];
+          }
+        }
+      }
+
+      switch (eventtitle)
+      {
+       case "JSREADY":
+        for (i = 0; i < actionscriptCommandQueue.length; i++) {
+          this.callToFlash(actionscriptCommandQueue[i][0],actionscriptCommandQueue[i][1]);
+        }
+        break;
+       case "PAUSED":
+        raisePauseEvent();
+        break;
+       case "BUFFERING":
+        newController.notify(newController.EVENTS.BUFFERING);
```

Comment:
out of curiosity - why are all other handlers in a function and this one is not?